cmd/controller/run.go

```diff
@@ -79,12 +83,27 @@ func Run(opts Options, runLog logr.Logger) {
 		os.Exit(1)
 	}
 
+	pkgClient, err := pkgclient.NewForConfig(restConfig)
+	if err != nil {
+		runLog.Error(err, "building app client")
+		os.Exit(1)
+	}
+
 	appFactory := AppFactory{
 		coreClient: coreClient,
 		kcConfig:   kcConfig,
 		appClient:  kcClient,
 	}
 
+	server, err := apiserver.NewAPIServer(restConfig)
+	if err != nil {
+		runLog.Error(err, "creating server")
+		os.Exit(1)
+	}
+	server.Run()
+
+	// TODO: we may need to sleep here to give the server time to start up
```

Comment:
How common of an issue is this? Is there any kind of wait we can put in place for this?

Reply:
I have yet to hit it locally, so I didn't put any kind of waiting behavior in there. I think if we want to be safe we may be able to add something as simple as a couple second sleep.